internal/host/host.go

```diff
@@ -1412,3 +1417,49 @@ func (m *Manager) HandleReclaimBootArtifactDownload(ctx context.Context, h *mode
 func (m *Manager) HandleReclaimFailure(ctx context.Context, h *models.Host) error {
 	return m.sm.Run(TransitionTypeReclaimFailed, newStateHost(h), &TransitionArgsUnbindHost{ctx: ctx, db: m.db})
 }
+
+// autorenameHost renames a host if it is necessary and possible. For example, if the name is
+// localhost.localdomain and the MAC address of the main network interface is A8:CD:16:AE:79:01 it
+// will be renamed to a8-cd-16-ae-79-01. Returns a boolean indicating if the host was actually
+// renamed.
+func (m *Manager) autorenameHost(host *models.Host, inventory *models.Inventory) bool {
+	if host.RequestedHostname != "" {
+		return false
+	}
+	if inventory == nil || inventory.Hostname == "" {
+		return false
+	}
+	if !funk.Contains(hostutil.ForbiddenHostnames, inventory.Hostname) {
+		return false
+	}
+	if len(inventory.Interfaces) < 1 {
+		return false
+	}
+	var selectedIf *models.Interface
+	for _, candidateIf := range inventory.Interfaces {
+		if candidateIf.MacAddress != "" {
```

Comment:
It might be possible for two hosts to have the same mac address https://issues.redhat.com//browse/MGMT-11888

Reply:
Thanks for pointing that out @CrystalChun! In the new version of the patch I added logic to select as main interface the first one that has both a MAC address and an non-local (not 127.0.0.1 or ::1) IP address. I think this should address this issue. I believe it is not necessary to add additional information to the name to make it unique, because MAC addresses of interfaces that are actually connected to the network should be unique. Please take another look and let me know what you think.

Reply:
Sounds good! Reviewed and had one comment. Would also ask @mkowalski to take a look too!

Reply:
Thanks, looks safe after the change. The bug in subject was exposed for NICs that were using link-local addresses and those will return `false` when fed into `net.IsGlobalUnicast`

Reply:
Thanks for taking a look @mkowalski!